### PR TITLE
add tags for heading

### PR DESCRIPTION
**Pull request by b232wang**

### CODE REVIEW

**Comment by b232wang** [comment]
It's like:
***\* TODO [#A] COMMENT Title :tag:a2%:
tags for this is ['tag', 'a2%'] 
the last colon is optional.

File: lib/parser.js [hunk 1]
```diff
@@ -28,20 +28,31 @@ const blockOption = seq(
...
 const blockOptions = blockOption.many();
 const startOfHeading = startOfLine.then(lexeme(regex(/[\*]+/)));
+
+const noSpaceText = lexeme(regex(/[^:^\n]+/));
+const atom = lexeme(regex(/[a-z0-9%@#_]+/i));
+const tag = lexeme(regex(/[\s]*:/)).then(atom);
+const ttl = lexeme(regex(/[^:]*/));
+const tagForm = ttl.then(tag.many()).skip(colon.times(0,1));
+const tagList = lazy(() =>
+                     { return tagForm; });
```
Comment (yorkie) [inline lib/parser.js]: Put L44-45 to only one line.

File: lib/parser.js [hunk 2]
```diff
@@ -28,20 +28,31 @@ const blockOption = seq(
...
+
 const heading = seq(
   startOfHeading, singleLineText
 ).map(
   (val) => {
+    let tag = tagList.parse(val[1]).value || [];
```
Comment (yorkie) [inline lib/parser.js]: @b232wang may I ask you to create the `tagList` parser to a separate file like `tags-parser` and requires in this root file?

**Comment by yorkie** [comment]
@b232wang this patch looks pretty simple and readable, awesome works and LGTM but the above 2 small nits to improve the structuring :-)

File: lib/parser.js (hunk 2 above)
Comment (b232wang) [inline lib/parser.js]: @yorkie Do you means that write tags parser to an other separate file then we can just find tags without `parser.js`? Also, tags cannot be filed of `OutLineNode`?

**Comment by b232wang** [comment]
@yorkie  Thanks, :). In `org-syntax.org`, It only mention `tags` on heading line, is that means we don't need support `tags` to `Affiliated Keywords` ?

**Comment by yorkie** [comment]
Hey @scottoasis How do you think about @b232wang's last comment :-)

**Comment by scottoasis** [comment]
@b232wang According to the [Official Syntax Specification](https://github.com/weflex/node-orgmode/blob/master/org-syntax.org), _Affiliated Keywords_ are level-local key-val configuration lines start with a hash-plus (`#+`) and its **key** and **value** separated by a colon (`:`).

In short, Keywords are not Headlines and thus they don't have _Tags_.

**Comment by b232wang** [comment]
@scottoasis So, is that means only `heading` may have `tags` ? Should we support a function to find `heading` by using `tags`?
eg: doc.findByTags(str1)[0]

**Comment by scottoasis** [comment]
@b232wang 

> So, is that means only heading may have tags ?

Absolutely. _Tags_ is a _Headline_-Only feature.

> Should we support a function to find heading by using tags?

As this question falls down to the requested feature list (if any) of the next shipping, hence @yorkie knows better.

Yet in general, this could be an interesting feature to have as Emacs _Org Mode_ also supports _Sparse Tree Searching_. You can find out more about Sparse Tree Searching in [Org Mode Manual](http://orgmode.org/manual/Tag-searches.html#Tag-searches).

Thanks.

**Comment by yorkie** [comment]
> Should we support a function to find heading by using tags?

@b232wang in this pr, shouldn't

**Comment by yorkie** [comment]
LGTM and merged this PR, BTW @b232wang please use branch instead of forking to submit your PR next time :-)

**Pull request merged**